src/xp.runner.test/CommandLineTest.cs

```diff
@@ -74,6 +82,44 @@ public void ar(string arg)
             Assert.IsType<Ar>(new CommandLine(new string[] { arg }).Command);
         }
 
+        [Fact]
+        public void plugin()
+        {
+            Assert.Equal("web", (new CommandLine(new string[] { "web" }).Command as Plugin).Name);
+        }
+
+        [Theory]
+        [InlineData(@"{""scripts"":{""serve"":""xp web org.example.web.App""}}")]
+        [InlineData(@"{""scripts"":{""serve"":""xp 'web' org.example.web.App""}}")]
+        [InlineData(@"{""scripts"":{""serve"":""xp web 'org.example.web.App'""}}")]
+        [InlineData(@"{""scripts"":{""serve"":""xp web 'org.example.web.App""}}")]
+        public void script_via_composer_file(string source)
+        {
+            var fixture = new CommandLine(new string[] { "serve" }, ComposerFile(source));
+            Assert.Equal("web", (fixture.Command as Plugin).Name);
+            Assert.Equal(new string[] { "org.example.web.App" }, fixture.Arguments);
+        }
+
+        [Fact]
+        public void script_via_composer_file_with_arguments()
+        {
+            var composer = ComposerFile(@"{""scripts"":{""serve"":""xp web org.example.web.App""}}");
+            Assert.Equal(
+                new string[] { "org.example.web.App", "dev" },
+                new CommandLine(new string[] { "serve", "dev" }, composer).Arguments
+            );
+        }
+
+        [Theory]
+        [InlineData(@"{}")]
+        [InlineData(@"{""scripts"":{}}")]
+        [InlineData(@"{""scripts"":{""serve"":""xp web org.example.web.App""}}")]
+        [InlineData(@"{""scripts"":{""test"":""phpunit""}}")]
```

Comment:
There should also be a test for a command that starts with 'xp', but is not xp, like 'xpcli'

Reply:
Added [xprop](https://linux.die.net/man/1/xprop) as an example.